### PR TITLE
Fixes Brave ads database migration from schema 10

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/17231

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See https://github.com/brave/brave-browser/issues/17231

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/database/database_table_util.cc [hunk 1]
```diff
@@ -70,7 +70,7 @@ std::string BuildInsertQuery(const std::string& from,
   const std::string comma_separated_to_columns =
       base::JoinString(to_columns, ", ");
 
-  return base::StringPrintf("INSERT INTO %s (%s) SELECT %s FROM %s %s",
+  return base::StringPrintf("INSERT INTO %s (%s) SELECT %s FROM %s %s;",
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/database/database_table_util.cc:73]: Fixes query bug due to missing semi-colon

File: vendor/bat-native-ads/src/bat/ads/internal/database/tables/conversion_queue_database_table.cc [hunk 2]
```diff
@@ -340,16 +340,38 @@ void ConversionQueue::MigrateToV10(DBTransaction* transaction) {
...
+                                            "creative_set_id",
+                                            "creative_instance_id",
+                                            "advertiser_id",
+                                            "conversion_id",
+                                            "timestamp"};
+
+  util::Migrate(transaction, get_table_name(), temp_table_name,
+                columns, true);
+
+  util::Rename(transaction, temp_table_name, get_table_name());
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/database/tables/conversion_queue_database_table.cc:374]: Migrates and adds a new column

File: vendor/bat-native-ads/src/bat/ads/internal/database/tables/conversions_database_table.cc [hunk 3]
```diff
@@ -247,7 +247,7 @@ void Conversions::MigrateToV1(DBTransaction* transaction) {
   CreateIndexV1(transaction);
 }
 
-void Conversions::MigrateToV11(DBTransaction* transaction) {
+void Conversions::MigrateToV10(DBTransaction* transaction) {
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/database/tables/conversions_database_table.cc:250]: This table version should not have changed

**Review by emerick — APPROVED** [review]
LGTM

**Review by moritzhaller — APPROVED** [review]
LGTM

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.29.46 Chromium: 92.0.4515.115 (Official Build) nightly (64-bit)
--- | ---
Revision | 48cb2f4029b84b003719740a6cf9ca73f374a857-refs/branch-heads/4515_105@{#4}
OS | Windows 10 OS Version 2009 (Build 19042.1110)
```

Upgraded from `1.24.82 Chromium: 90.0.4430.93` to `1.29.46 Chromium: 92.0.4515.115` and ensured the following:


Ensured that `ad rewards` reconciled without any issues and the unlinded tokens were retrieved as per the following:

```
[20744:20048:0806/184933.936:VERBOSE1:ad_rewards.cc(250)] Successfully reconciled ad rewards
[24456:23824:0803/183153.803:VERBOSE1:refill_unblinded_tokens.cc(301)] Added 50 unblinded tokens, you now have 50 unblinded tokens
```

Ensured that both `veiw` & `clicked` worked as expected for `NTP SI` as per the following:

```
[25884:13192:0803/183821.911:VERBOSE1:confirmations.cc(250)] Successfully redeemed unblinded token with confirmation id 91979b19-23f3-4cf9-a7bf-490623cae49f, creative instance id e3af9ae9-b0a9-406d-b98f-ed506734aba7 and view
[25884:13192:0803/183821.911:VERBOSE1:confirmations.cc(273)] Added 1 unblinded payment token with an estimated redemption value of 0.01 BAT, you now have 2 unblinded payment tokens
[25884:13192:0803/183821.911:VERBOSE1:refill_unblinded_tokens.cc(66)] No need to refill unblinded tokens as we already have 48 unblinded tokens which is above the minimum threshold of 20
```

```
[25884:13192:0803/184018.534:VERBOSE1:confirmations.cc(250)] Successfully redeemed unblinded token with confirmation id 19fc4ec9-fd8c-4d0e-819c-a3a1cf74afc4, creative instance id e3af9ae9-b0a9-406d-b98f-ed506734aba7 and click
[25884:13192:0803/184018.535:VERBOSE1:confirmations.cc(273)] Added 1 unblinded payment token with an estimated redemption value of 0 BAT, you now have 3 unblinded payment tokens
[25884:13192:0803/184018.535:VERBOSE1:refill_unblinded_tokens.cc(66)] No need to refill unblinded tokens as we already have 47 unblinded tokens which is above the minimum threshold of 20
```

Ensured that website classification is working after upgrading as per the following:

* `[25884:13192:0803/184146.261:VERBOSE1:text_classification_processor.cc(62)] Classified text with the top segment as technology & computing-software` - https://www.brave.com
* `[25884:13192:0803/184319.870:VERBOSE1:text_classification_processor.cc(62)] Classified text with the top segment as technology & computing-technology & computing` - https://videocardz.com
* `[25884:13192:0803/184543.369:VERBOSE1:text_classification_processor.cc(62)] Classified text with the top segment as gaming-gaming` - https://www.metacritic.com

Ensured that ad notifications are working as per the following:

```
[25884:13192:0803/184419.694:VERBOSE1:ad_notification_serving.cc(222)] Serving ad notification:
  uuid: 173d9559-c732-4fbd-96d1-4b1bc611ec98
  creativeInstanceId: 293642dd-9dae-4042-8232-f2db71e318fc
  creativeSetId: e3fa4776-d529-4ae9-9c6e-7741232ee1df
  campaignId: 399a5816-33b9-4167-b7c6-d8783ed9ddd8
  advertiserId: b6d581f3-cf2c-4422-a43c-75d4a675fe58
  segment: personal finance
  title: Bitbuy - Get Crypto In Canada
  body: Get a $20 free credit when you make your first deposit.
  targetUrl: https://bitbuy.ca/en/sign-up/?c=Brave
```

Moved the date ~3 days ahead and ensured that Brave successfully redeemed the unblinded payment tokens as per the following:

```
[20744:20048:0806/185033.946:VERBOSE1:account.cc(195)] Successfully redeemed unblinded payment tokens
```

Also ensured that:

* ensured that `Ads received this month` is correctly being updated under `brave://rewards`
* ensured that `Current earnings this month (estimated)` is correctly being updated under `brave://rewards`
* ensured that `30-day ads history` is correctly being updated under `brave://rewards`

**Pull request merged**